src/runner.py

```diff
@@ -56,29 +161,49 @@ def virt_type(self):
                 if label["name"] in {"container", "virtual-machine"}
             )
             return next(remote_virt_type, None)
+        return None
 
 
 class RunnerError(Exception):
-    pass
+    """Generic runner error."""
 
 
-class RunnerCreateFailed(RunnerError):
-    pass
+class RunnerCreateError(RunnerError):
+    """Error for runner creation failure."""
 
 
-class RunnerRemoveFailed(RunnerError):
-    pass
+class RunnerRemoveError(RunnerError):
+    """Error for runner removal failure."""
 
 
-class RunnerStartFailed(RunnerError):
-    pass
+class RunnerStartError(RunnerError):
+    """Error for runner start failure."""
 
 
 class RunnerManager:
+    """Class for querying and controlling the runners.
+
+    Attributes:
+        proxies (Dict[str, str]): Mapping of proxy env vars.
```

Comment:
Would it be worth defining these as typed attributes on the class?

Reply:
After some throught, attributes are meant to be publicly accessible. So would it be better if they are typed in the docstring? So someone using the class only need to read the docstrings and function signature to know how to use the class?

Reply:
Does `mypy` read the types from the docstring?

Reply:
No, but types are inferred from the input of the `__init__` and functions called.